Spark: Don't daemonize, don't pretend to tail logs we don't use

zmerlynn opened the pull request:
Along the way: Fixup port mismatch in worker container

zmerlynn committed:
Spark: Don't daemonize, don't pretend to tail logs we don't use

Along the way: Fixup port mismatch in worker container

zmerlynn commented:
cc @timstclair (yes, I meant to assign to other Tim, since this is addressing https://github.com/kubernetes/kubernetes/pull/16320#discussion_r43505013)

zmerlynn commented:
Ping. :)

zmerlynn commented:
@k8s-bot unit test this

timothysc commented on examples/spark/images/master/start.sh:
```diff
@@ -17,8 +17,6 @@
-/opt/spark/sbin/start-master.sh
-tail -F /opt/spark/logs/*
+# Run spark-class directly so that when it exits (or crashes), the pod restarts.
+/opt/spark/bin/spark-class org.apache.spark.deploy.master.Master --ip spark-master --port 7077 --webui-port 8080
```
Did you verify the output works.. I tried building your images but they ubuntu based :-(

zmerlynn replied:
Yup. There's a pile of private iterations hidden behind that one tag bump.

zmerlynn replied:
What failed? Blech, I thought this was supposed to be agnostic.

timothysc replied:
apt-get failed on the base image.

timothysc replied:
More to the point, do you need to install scala?  You're downloading the compiled binaries, which should only need the jvm iirc.

timothysc replied:
Come to think of it why are you installing scala?  

You should only need the jvm for compiled binaries.

zmerlynn replied:
That's a bit of legacy I haven't even touched. I can try to take it out, gimme a bit.

zmerlynn replied:
Done. I took out `setup_client.sh` as well, because it's no longer how we get to the master. (It used to shim in the address for the driver, if I'm understanding it, now DNS just works.)

timothysc commented:
So slightly beyond the scope of the original ask, but fine by me.  Minor comments otherwise lgtm 

Broader question is going to be, should the images live in this repo?  I'm going to say they probably don't.

zmerlynn commented:
Right. I was commenting to @timstclair (who reviewed the last one) that I really did mean to assign it to @timothysc and didn't just typo it, because I was addressing your comments. :)

zmerlynn committed:
Remove unneeded pieces

zmerlynn commented:
PTAL. You should be able to build now.

timothysc commented:
@zmerlynn lgtm for now, but as config starts to become available we should leverage and move config out.  

/cc @pmorie

zmerlynn commented:
@timothysc: Cool, self-applying.
